JSON:API-RAML-Osprey-Express article

postatum opened the pull request:
See this for details https://github.com/postatum/raml-playground/blob/raml-osprey-expressjs/raml-osprey-expressjs/README.md

postatum committed:
Create structure for new article

postatum committed:
Add simple example from Osprey

postatum committed:
Add package.json

postatum committed:
Define initial json-api RAML spec

postatum committed:
Convert json examples to yaml

postatum committed:
Define strict resource types

postatum committed:
Remove strict resource types

postatum committed:
Prepare js example base

postatum committed:
Convert examples back to json

Trying to make Osprey and RAML parser
digest the api.raml

postatum committed:
Simplify included RAML json-api libs

postatum committed:
Fix few mistakes in raml libs and spec

postatum committed:
Rework json-api RAML libs to work with Osprey

postatum committed:
Implement osprey+express app

postatum committed:
Write server interaction readme

postatum committed:
Finish writing doc

jstoiko committed:
Update README.md

jstoiko commented on raml-osprey-expressjs/README.md:
```diff
@@ -0,0 +1,172 @@
+*pokemon_api/libraries/jsonApiLibrary.raml* - Defines RAML Data Types required to satisfy JSON API requirements.
+*pokemon_api/libraries/jsonApiCollections.raml* - Defines resourceTypes and traits required to properly describe the JSON API RAML spec.
+
+## Things changed in JSON API RAML lib
```
This section/paragraph can be moved to `libraries/README.md`

postatum replied:
Fixed

jstoiko commented on raml-osprey-expressjs/pokemon_api/index.js:
```diff
@@ -0,0 +1,107 @@
+var express = require('express')
```
let's make sure to write this file in modern ES6 style, e.g. using `const` for requires and immutable variables like PORT.

makes sense?

postatum replied:
Yes. Fixed. Now using `const` for everything that is not reassigned and `let` in loops.

jstoiko committed:
Update package.json

postatum committed:
Fix var/let/const usage

postatum committed:
Fixed types param. Add $ to sh commands

jstoiko committed:
refactor/cleanup

jstoiko committed:
more cleanup

jstoiko committed:
more refactor/renaming

jstoiko committed:
rewrite (WIP)

postatum committed:
Merge pull request #2 from jstoiko/postatum/raml-osprey-expressjs

refactor/cleanup

postatum committed:
Refactor middleware usage. Drop media params in resp

postatum committed:
Return 415 to requests with media params

postatum committed:
Move middleware and helpers to separate modules

postatum committed:
Rework header param check

postatum committed:
Reject on invalid Accept

postatum committed:
Remove hand-made Accept header rejection

Looks like Express already requires one of Accept header
values be clean application/vnd.api+json

postatum committed:
Wrap Express errors in JSON API format

postatum committed:
Fix html 500 code generated by middleware

postatum committed:
Implement Osprey validation errors wrapper

postatum committed:
Use forked osprey-method-handler with JSON API support

postatum committed:
Add self links to responses

postatum committed:
Make PokemonAttrs props optional to make PATCH work

jstoiko commented:
Moved to a [branch](https://github.com/raml-org/raml-playground/tree/postatum/raml-osprey-expressjs).
